packages/lib/src/interactions/svg/SvgRect.tsx

```diff
@@ -12,14 +12,13 @@ function SvgRect(props: Props) {
   const { coords, strokePosition, ...svgProps } = props;
 
   const [start, end] = coords;
-  const { stroke, strokeWidth = 1 } = svgProps;
+  const { strokeWidth = 1 } = svgProps;
 
   // Shrink/grow rectangle to simulate positioning stroke inside/outside
   // https://stackoverflow.com/questions/7241393/can-you-control-how-an-svgs-stroke-width-is-drawn
-  const offset =
-    stroke && strokePosition
-      ? (strokeWidth / 2) * (strokePosition === 'inside' ? 1 : -1)
-      : 0;
+  const offset = strokePosition
+    ? (strokeWidth / 2) * (strokePosition === 'inside' ? 1 : -1)
+    : 0;
```

Comment:
There was a regression here: since I've replaced the `stroke` prop with the `stroke` CSS property in `SelectToZoom`, the `offset` logic above was no longer working (i.e. `offset` was always `0`). As a fix, I remove `stroke` from the condition, assuming that if the consumer passes prop `strokePosition`, it's because the `SvgRect` has a stroke... This works, but a similar problem occurs if the stroke width is set via CSS... Not sure how to fix this... Reading the styles dynamically is requires rendering the component once first, which complexifies things... Instead of `strokePosition`, I could just ask for the `offset` directly, but then the user may forget to update the prop when adjusting the stroke width... Any thoughts?

Reply:
There's indeed a CSS-only solution to the inside-stroke problem: it consists in doubling up the stroke and using `clip-path` to crop it and keep only the part inside the rectangle. However, this introduces an implicit coupling: users have to know and remember to double up the stroke width. For outside strokes, there's a solution with `mask`, but it hides the inside of the rectangle, so it's not compatible with `fill`.

It's a complex problem to solve, so I'm not fond of the idea of consumers having to deal with it themselves. Since I can't come up with a perfect solution, I say we just go with a convention: don't use a CSS property if there's an equivalent SVG attribute.

- I've now re-added `stroke &&` to the condition like it was before.
- I've put back all the props on `SvgRect` that I had moved to CSS, in `SelectToZoom` and `AxialSelectToZoom`.
- I've kept only the `transition` property in the CSS, with a single `selection` class.